### PR TITLE
Drivetrain abstractor

### PR DESCRIPTION
**Pull request by ManiacMoustache**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Steps to Test
1. Get a robot with a drivetrain
2. Enable an autonomous component that uses the drivetrain during teleop
3. Use the drive controller while the component is running

If you can drive the robot while the component is running, it works

## Changelog

We had a poor way of handling who gets the drivetrain.

## What?

Added a new component to act as a controller of who gets control of the drivetrain. Right now it's implemented in all autonomous components that use the drivetrain. It's not as complicated as it could be, but there's always the potential for more later.

## Why?
Because previously we'd have issues with who controls the drivetrain

## Automation
(We don't have an issue for this)

### CODE REVIEW

File: components/turnToAngle.py [hunk 1]
```diff
@@ -74,9 +74,9 @@ def turn(self):
         """Turns the robot based off of the speed determined in setSpeedFunc"""
         self.setSpeedFunc()
         if self.change > 0:
-            self.driveTrain.setTank(-1 * self.speed, self.speed)
+            self.driveTrainHandler.setTank(self, -1 * self.speed, self.speed)
```
Comment (ChrisHirsch) [inline components/turnToAngle.py:77]: Still hardcoded to setTank? This should be abstracted now right?

File: components/turnToAngle.py [hunk 2]
```diff
@@ -74,9 +74,9 @@ def turn(self):
...
         else:
-            self.driveTrain.setTank(self.speed, -1 * self.speed)
+            self.driveTrainHandler.setTank(self, self.speed, -1 * self.speed)
```
Comment (ChrisHirsch) [inline components/turnToAngle.py:79]: Still hardcoded to setTank? This should be abstracted now right?

File: components/turnToAngle.py (hunk 1 above)
Comment (ChrisHirsch) [reply components/turnToAngle.py:77]: If we need the correct handler, considering abstracting that and passing it in?

File: robot.py [hunk 3]
```diff
@@ -68,10 +69,12 @@ class MyRobot(MagicRobot):
     goToDist: GoToDist
     ballCounter: BallCounter
     colorSensor: ColorSensor
+    driveTrainHandler: DriveTrainHandler
 
     # Test code:
     testBoard: TestBoard
 
+    controllerDeadzone = tunable(.06)
```
Comment (ChrisHirsch) [inline robot.py:77]: Comment on what this is?

**Review by ChrisHirsch — CHANGES_REQUESTED** [review]
Let me know if you need help with the abstraction for tank (or any specific drive)

**Review by hell0w0rld2 — APPROVED** [review]
👍

**Review by hell0w0rld2 — APPROVED** [review]
👍

**Review by ChrisHirsch — APPROVED** [review]
Yep..it's a good start! +1

**Pull request merged**